lib/topological_inventory/sync/inventory_upload/validator_worker.rb

```diff
@@ -0,0 +1,58 @@
+require "json"
+require "topological_inventory/sync/worker"
+require "topological_inventory/sync/inventory_upload/parser"
+
+module TopologicalInventory
+  class Sync
+    module InventoryUpload
+      class ValidatorWorker < Worker
+        include Logging
+
+        def queue_name
+          "platform.upload.topological-inventory"
+        end
+
+        def persist_ref
+          "topological-inventory-upload-validator"
+        end
+
+        def perform(message)
+          payload = JSON.parse(message.payload)
+
+          account, request_id, payload_id = payload.values_at("account", "request_id", "payload_id")
+          log_header = "account [#{account}] request_id [#{request_id}]"
+
+          logger.info("#{log_header}: Validating payload [#{payload_id}]...")
+
+          inventory = Parser.parse_inventory_payload(payload["url"])
```

Comment:
@agrare does this parse the whole payload? Are we planning to do a full validation here? Or just header validation?

Reply:
It does parse the full payload but I'm looking at ways to break out once we reach the collections key